be/src/storage/rowset/struct_column_iterator.cpp

```diff
@@ -118,8 +135,19 @@ Status StructColumnIterator::next_batch(const SparseRange& range, Column* dst) {
         down_cast<NullableColumn*>(dst)->update_has_null();
     }
     // Read all fields
+    size_t row_count = 0;
+    auto fields = struct_column->fields();
+    for (int i = 0; i < _field_iters.size(); ++i) {
+        if (_access_flags[i]) {
+            RETURN_IF_ERROR(_field_iters[i]->next_batch(range, fields[i].get()));
+            row_count = fields[i]->size();
+        }
+    }
+
     for (int i = 0; i < _field_iters.size(); ++i) {
-        RETURN_IF_ERROR(_field_iters[i]->next_batch(range, struct_column->fields()[i].get()));
+        if (!_access_flags[i]) {
+            fields[i]->append_default(row_count);
+        }
```

Comment:
would it be removed directly?

Reply:
can't now, we doesn't unpack column on storage now